required/latex-lab/latex-lab-sec.dtx

```diff
@@ -280,131 +284,289 @@
+% The second argument of this socket will pass the normal \cs{@hangfrom} command
+% if tagging is not active. It is not used with tagging.
+% The first argument passes four brace groups:
+% the level, a boolean for \enquote{nonumber}, 
+% the actual content of the number and the title.
+%
+% Attention: The socket opens a structure that it doesn't close (it is closed by the \cs{par}).
+% It therefore does not handle the full tagging of the title. In a new implementation
+% of the sectioning command this will perhaps have to change.
 %    \begin{macrocode}
-        \tl_set:Nn\l_@@_para_main_tag_tl {NonStruct}
-        \tl_set:Nn\l_@@_para_tag_tl {Span}
-     }
-    ,sec-stop-chapter .meta:n = { sec-stop-part}
-    ,sec-start .code:n = % #1 is a name like "section"
-      {
-        \bool_if:NT\l_@@_sec_Sect_bool
-          {
-            \@@_sec_end:n    {\cs_if_exist_use:c{toclevel@#1}+0}
-            \@@_sec_begin:nn {\cs_if_exist_use:c{toclevel@#1}+0}{tag=Sect}
-          }
-        \tl_set:Nn\l_@@_para_tag_tl{#1}
-      }
-    ,sec-start .value_required:n = true
-    ,sec-split-para .code:n = {\@@_tool_para_split:}
-    ,restore-para .code:n = {\@@_setup_restore_para_default:}
-    ,sec-stop .code:n =
-      {
-        \par\@@_sec_end:n   {\cs_if_exist_use:c{toclevel@#1}+0}
-      }
-    ,sec-stop .value_required:n = true
-    ,sec-add-grouping .bool_set:N = \l_@@_sec_Sect_bool
+\NewTaggingSocketPlug{sec/title/hang}{kernel}
+  {
+    \@@_set_title_hang:nNnn #1
   }
+\AssignTaggingSocketPlug{sec/title/hang}{kernel}
+%    \end{macrocode}
+%
+% The \texttt{sec/title/init} socket is used to initialize in the tagging of titles
+% for sectioning commands that are setup with \cs{@startsection}. The tagging
+% makes use of the para-tagging, so we change the tag (for the
+% cases where it is not set explicitly) and flatten the paragraph.
+% It takes 1 argument, the level.
+%    \begin{macrocode}
+\NewTaggingSocketPlug{sec/title/init}{kernel}
+ {
+   \tl_set:Ne\l_@@_para_tag_tl{\UseStructureName{sec/#1/title}}
+   \bool_set_true:N \l_@@_para_flattened_bool
+ }
+\AssignTaggingSocketPlug{sec/title/init}{kernel}
 %    \end{macrocode}
 %
+% This socket handles the tagging between a run-in heading and the following text.
+% It takes no argument.
+%    \begin{macrocode}
+\NewTaggingSocketPlug{sec/title/split}{kernel}
+ {
+   \@@_sec_title_split:
+ }
+\AssignTaggingSocketPlug{sec/title/split}{kernel}
+%    \end{macrocode}
+%
+% The following tagging socket command is used to handle the tagging
+% of the number in the title of run-in headings. Similar to the hang variant it does not
+% handle the full structure but relies in part on the paragraph
+% tagging. This again can change if sectioning commands are reimplemented.
+% It takes as first argument the level and a boolean for the numbering.
+% The second argument contains the formatted number (if numbered) and the destination.
+%    \begin{macrocode}
+\NewTaggingSocketPlug{sec/title/runin}{kernel}
```

Comment:
I find the name somewhat odd. it says "title/runin"  so far so good but then it deals only with the heading number???

If this is a socket that we don't really epxect to survive then perhaps do not define it in lttagging but only here and drop it once there is a different implementation.